dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java

```diff
@@ -231,14 +231,14 @@ private void refreshInvoker(List<URL> invokerUrls) {
             }
             // use local reference to avoid NPE as this.cachedInvokerUrls will be set null by destroyAllInvokers().
             Set<URL> localCachedInvokerUrls = this.cachedInvokerUrls;
-            if (invokerUrls.isEmpty() && localCachedInvokerUrls != null) {
-
-                // 1-4 Empty address.
-                logger.warn(REGISTRY_EMPTY_ADDRESS, "configuration ", "",
-                    "Service" + serviceKey + " received empty address list with no EMPTY protocol set, trigger empty protection.");
-
-                invokerUrls.addAll(localCachedInvokerUrls);
+            if (invokerUrls.isEmpty()) {
+                if(localCachedInvokerUrls != null){
+                    // 1-4 Empty address.
+                    logger.warn(REGISTRY_EMPTY_ADDRESS, "configuration ", "",
+                        "Service" + serviceKey + " received empty address list with no EMPTY protocol set, trigger empty protection.");
 
+                    invokerUrls.addAll(localCachedInvokerUrls);
+                }
```

Comment:
```suggestion
            if (invokerUrls.isEmpty()) {
                if (localCachedInvokerUrls != null && !localCachedInvokerUrls.isEmpty()) {
                    // if the previous invokerUrls is not empty, warn and use the previous invokerUrls instead.
                    
                    // 1-4 Empty address.
                    logger.warn(REGISTRY_EMPTY_ADDRESS, "configuration ", "",
                        "Service" + serviceKey + " received empty address list with no EMPTY protocol set, trigger empty protection.");

                    invokerUrls.addAll(localCachedInvokerUrls);
                } // else, if the previous invokerUrls is empty, do nothing.
            } else {
                // cache if invokerUrls is not empty
                localCachedInvokerUrls = new HashSet<>();
                localCachedInvokerUrls.addAll(invokerUrls);//Cached invoker urls, convenient for comparison
                this.cachedInvokerUrls = localCachedInvokerUrls;
            }
```

Please add some comment

Reply:
Agree with @AlbumenJ 
A small suggestion:  use `CollectionUtils.isNotEmpty` to judge empty